src/modules/contactForm/pages/ContactForm/ContactForm.spec.tsx

```diff
@@ -1,19 +1,47 @@
 import React from 'react'
 import { shallow } from 'enzyme';
 
-import ContactForm from './ContactForm';
+import { ContactForm } from './ContactForm';
+import { steps } from '../../utils/ContactFormUtils'
 
 describe('ContactForm', () => {
-    let props, wrapper;
+    let props, wrapper
 
     beforeEach(() => {
         props = {
-            listenForUser: jest.fn()
+            listenForUser: jest.fn(),
+            submitContactForm: jest.fn(),
+            classes: jest.enableAutomock()
         };
         wrapper = shallow(<ContactForm {...props} />);
     });
 
     it('should match snapshot', () => {
-        expect(wrapper).toMatchSnapshot();
+        expect(wrapper).toMatchSnapshot()
     });
-});
\ No newline at end of file
+
+    describe('control functions', () => {
+
+        it('should handle next correctly', () => {
+            wrapper.setState({ activeStep: 1 })
+            wrapper.instance().handleNext()
+
+            expect(wrapper.state('activeStep')).toBe(2)
+        });
+
+        it('should submit form on last next', () => {
+            wrapper.setState({ activeStep: (steps.length - 1) })
+            wrapper.instance().handleNext()
+
+            expect(wrapper.state('activeStep')).toBe(steps.length - 1)
+            expect(props.submitContactForm).toBeCalled()
+        });
+
+        it('should change step on back', () => {
+            wrapper.setState({ activeStep: 1 })
+            wrapper.instance().handleBack()
+
+            expect(wrapper.state('activeStep')).toBe(0)

```

Comment:
do you think we should test the edge cases of a `handleBack()` call when the state is `0`? the UI will never get there, but it would provide a nice test to make sure we don't accidentally shoot ourselves in the foot